internal/flow/internal/controller/node_builtin_component.go

```diff
@@ -165,11 +165,13 @@ func NewBuiltinComponentNode(globals ComponentGlobals, reg component.Registratio
 
 func getManagedOptions(globals ComponentGlobals, cn *BuiltinComponentNode) component.Options {
 	cn.registry = prometheus.NewRegistry()
+	parent, id := path.Split(cn.globalID)
```

Comment:
Note that from the internal discussion about this, there was a mention to make `component_path` rooted to make it clearly indicate a global path:

> Global IDs start with a path separator, similarly to absolute file paths, to clearly indicate rooting.

Additionally, if there is no parent, `component_path` should be `/`: 

> component_path: the global ID of the containing subgraph for a component. If there is no containing subgraph, component_path is the rooted path /.

(This rooting logic applies to the other metrics as well wherever the labels are being split)

Reply:
I do like the leading `/` too.